### DISCUSSION TITLE
Cannot get VLC to view stream

### DISCUSSION THREAD

**Discussion by PeteSki1618** [GerryDazoo/Slinger#291] [General]
This type of configuration is WAY above my expertise.  However, I have successfully configured the config.ini file, ran the EXE program (results below) which finds the Slingbox Pro (I noticed some people had issues after the shutdown but mine existed before, as well).  When I run VLC using the Open Network Stream, Network Protocol http://192.168.0.254/slingbox and click Play, nothing happens.  If I make changes to the Config.ini file I get an error message but at present I get no errors but no video.  Also, no matter what password I put in, I do not get any changes (no errors) so is there a way to see if the password is incorrect?  I'm pretty sure I have the correct one but several were listed when I ran your script but none give an error.  Any help would be greatly appreciated.  I am lost without my Slingbox.  Below is the stream from the EXE:

Version : 3.08d Running on Windows-10-10.0.22621-SP0 pid= 11072
Using config file config.ini
Connection Manager Running on port 5001 with 10 max streams using URL slingbox.
Streamer Running:  10 config.ini SLINGBOX 5001 5001 8388608
starting up on  port 5001 8388608
Checking for slingbox at ('192.168.0.254', 5001)
('192.168.0.254', 5001) OK
5001 Using slingbox at  ('192.168.0.254', 5001)
Streamer:  5001 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off

**Comment by tom76017** [comment]
VLC needs the IP address of your server and port not the Slingbox.  http://_**IP-windowsPC**_:8080/slingbox

!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
Edit, my bad.   Both server and Slingbox are on port 5001.   http://_**IP-windowsPC**_:5001/slingbox.   I just copied the 8080 on the server from the original README.txt

**Comment by GerryDazoo** [comment]
Based on your config ( Why did you change the server port default from 8080 to 5001?

 http://IP-windowsPC:5001/slingbox

You'll know if your password is wrong when you try to start the first stream

**Comment by PeteSki1618** [comment]
Sorry, yes I did put  http://192.168.0.254:5001/slingbox (sorry for the mistype) but nothing happens at all.  I have been using port 5001 for many years with no problems so I just left it that way.  As you can see from the EXE results, it DOES find it at that address. 
 If you think I should change it to 8080, I can.  Thanks again for your help.

**Comment by GerryDazoo** [comment]
Well, the code isn't that smart. All it checks is there is something responding to connection requests at that address.
What port do you have in the config
[SERVER]
port=xxxx

192.168.0.254 is the IP address of the slingbox according to the logs. DO NOT use that in your streaming request. Use the IP address for the box running the server. You connect to the server not the slingbox.

Use xxxx in your streaming request.

**Comment by PeteSki1618** [comment]
Please pardon my ignorance.  As I said, I am very limited when it comes to this type of thing.  I do have port=5001 in the config.ini.  However, I was using the IP address of the Slingbox (254) in VLC.  I'm not sure what you mean by "the box running the server".  In my router I have listed the IP address of my computer, the Subnet Mask, Default Gateway, Primary and Secondary DNS Servers.  Do I use one of those?  By the way, I really appreciate your help and will gladly contribute if I can get it up and running.

**Reply by DoogieHowitzerWMD** [reply]
In the config.ini file, there are two ports you need to specify:

- the port that the slingbox will use to communicate with your server (within code in the [SLINGBOX] header, directly below where you specify the ip address of your slingbox, the default value is 5201). If you are using 5001 here, that should be fine.
- the port that you use to connect to your server to view a stream (within code in the [SERVER] header, the default value is 8080).

The second port number is what you need to put in the address to view the stream.

Also, when you type the address to watch your stream, it has to be the ip address of **the device running the server** (your computer, what Gerry mentioned as "the box running the server"), not the ip address of the slingbox. Per the output code you pasted, the slingbox is at 192.168.0.254, so the address for the server will not be that one.

**Reply by PeteSki1618** [reply]
Thank you VERY much for the clear explanation.  I now can access it using VLC but I keep getting a password error.  I tried every password listed in the script that I ran prior to the shutdown (there were 3 old ones listed but not the one I was using to log into my account) but none worked.  Is there any other way now to retrieve the password or reset the box?  Thanks again for all of your help!

**Reply by GerryDazoo** [reply]
Is it a Solo or Pro?

**Reply by PeteSki1618** [reply]
I have 2 HD Pro's, one called SlingboxHDPro and the other called LivingRoomProHD.  When I used to sign into my account, I had control of my two boxes and several friend's boxes.  I physically have in my possession my two boxes and another called Phil's Slingbox but am only connecting one at a time to see if I can get ANY of them to work.   I am only connecting ONE box at a time.  I am attaching the script results below.  When I ran the script to retrieve my passwords as seen below, I tried each of the passwords highlighted in red on the attachment (but erased for security).  Strangely enough, none of these were the passwords used to sign into my account, which I also tried with no luck.  Here is what the script looked like:
![PasswordsXX](https://user-images.githubusercontent.com/117535328/201764260-0bc48b9e-99fb-4407-a1db-de48534fd15f.jpg)

After running VLC, I get an error about the password in the Slingbox.exe window.  Here are the results.  Notice it says to check my password.   Thanks again for all of your help.  I am getting closer each time!

Version : 3.08d Running on Windows-10-10.0.22621-SP0 pid= 25324
Using config file config.ini
Connection Manager Running on port 8181 with 10 max streams using URL slingbox.
starting up on  port 8181 8388608
Streamer Running:  10 config.ini SLINGBOX 8181 8181 8388608
Checking for slingbox at ('192.168.0.254', 5001)
('192.168.0.254', 5001) OK
8181 Using slingbox at  ('192.168.0.254', 5001)
Streamer:  8181 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
11/14/2022, 15:50:02.328   Streaming request from ('192.168.0.100', 59906)

Slinginfo  "Solo/Pro/ProHD" 12 30 192.168.0.254 5001 3000 10
8181 Starting Stream for  192.168.0.100:59906
Connecting... ('192.168.0.254', 5001) Control
cmd: 0x67 err: 0x2b 8
8181 Error Starting Session. Check your admin password in config.ini file!
8181 ERROR: Slingbox session startup failed.
8181 Logging Out
EXCEPTION [WinError 10054] An existing connection was forcibly closed by the remote host 0x101 0 0x68 0 3 0 0x2000 0x0
Exception in thread Thread-2 (streamer):
Traceback (most recent call last):
  File "slingbox_server.py", line 372, in sling_cmd
    s_ctl.sendall( cmd )
ConnectionResetError: [WinError 10054] An existing connection was forcibly closed by the remote host

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "threading.py", line 1016, in _bootstrap_inner
  File "threading.py", line 953, in run
  File "slingbox_server.py", line 964, in streamer
    if s_ctl : s_ctl = closecontrol(s_ctl)
  File "slingbox_server.py", line 576, in closecontrol
    if Solo: sling_cmd( 0x68, b'')
  File "slingbox_server.py", line 375, in sling_cmd
    exit(1)
NameError: name 'exit' is not defined

**Reply by PeteSki1618** [reply]
By the way, since I had several password changes (none of which worked), I connected "Phil's Slingbox" which I know only had ONE password but get the same message.

**Reply by JoeM2016** [reply]
Try editing the config file to only connect to the first SB in your console log.  The console file looks normal until you try connecting the second one.

**Reply by PeteSki1618** [reply]
I am only connecting one SB at a time to see if I can get the correct password.  No matter which box I use, I get the same exact error message.

**Reply by tom76017** [reply]
Your json screen shows 5 or 6 adminPasswords.  Try each one, 1st without the remote activated. 
![image](https://user-images.githubusercontent.com/26171888/201768211-c2797135-85e1-4826-995c-4fab1cbf0676.png)

**Reply by PeteSki1618** [reply]
The first two adminPasswords are for my two boxes and both passwords are the same.  The 3rd and 4th are for friend's of mine and I don't have their boxes.  The last is another friend's box and I do have his box in my possession and tried his password on his box, too.  None of them work.  Also, my adminPassword is not the same as my login password when it used to be working but I tried the adminPasswords listed in the above script, I tried the userPassword (different than the adminPassword) and I tried my login password with no luck.  I KNOW my friend never changed his password on his box so I am almost positive his adminPassword is correct but I get the same error.  Thanks.

**Comment by jkaponis** [comment]
https://github.com/GerryDazoo/Slinger/files/9997979/Step-by-step.pdf

**Reply by PeteSki1618** [reply]
Thank you VERY much for that GREAT tutorial!  It really helped, especially for us "network challenged" people.  I now can access it using VLC but I keep getting a password error.  I tried every password listed in the script that I ran prior to the shutdown (there were 3 old ones listed but not the one I was using to log into my account) but none worked.  Is there any other way now to retrieve the password or reset the box?  Thanks again for all of your help!

**Reply by JoeM2016** [reply]
Where are you getting the password error? Slinger Console or VLC console?

**Reply by jkaponis** [reply]
https://github.com/GerryDazoo/Slinger/discussions/228#discussioncomment-4109510 change the username for your username

**Reply by PeteSki1618** [reply]
I'm getting the error in the Slinger Console (see Reply to GerryDazoo under DoogieH above).  
I don't see a "username" in the config.ini file, only "password".  Besides, I don't think I ever changed it from "admin".

**Comment by DoogieHowitzerWMD** [comment]
Do you know which of the three slingboxes is at the ip 192.168.0.254? From the file you have (what you showed in the screengrab) you are looking for the admin password. That password is not likely something you're familiar with or something you've used to connect to your slingboxes in the past.

Since you have three slingboxes on your network and three admin passwords in the file, you need to match the slingbox at 192.168.0.254 with it's associated admin password.

**Reply by PeteSki1618** [reply]
I am only connecting ONE box at a time and if I get it working, that is the only box I will use not to confuse things.  Both my "SlingboxProHD" and "Phil's Slingbox" are using 192.168.0.254 (again, not at the same time) and both are on port 5001.  The "LivingRoomProHD" is on 192.168.0.253 and port 5003.  I tried all passwords on all boxes.  However, these passwords WERE passwords that I set up at one time or another, not something obscure that was assigned to me.

**Comment by tom76017** [comment]
Turn off all Slingboxes except for the one your trying to connect to.  Confirm your Slingbox IP and your server IP (your PC).
Sample config.ini:

VVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVVV
[SLINGBOX]
sbtype="350/500/M1"       ; <<< pick the right SBtype
;sbtype="Solo/Pro/ProHD"
password=adminPassword12  ; <<< update this frpm the json values
ipaddress=192.168.0.254   ; <<< as seen from your router, make a static address
port=5001                 ; <<< or 5101, or 5201, or 5301  Did you capture this before EOL?
name=Slingbox-M1          ; <<< Anything

[SERVER]
port=8080                 ; Use this port in VLC
maxstreams=10
enableremote=no           ; Play with this after you get a video running

[REMOTE]
include=Remote.txt

**Reply by PeteSki1618** [reply]
Here's my config.ini file.  I'm sure all of the ports, etc, are correct for the box that I connect at that time (only one box at a time):

[SLINGBOX]
;sbtype="350/500/M1/M2"
sbtype="Solo/Pro/ProHD"
;sbtype="240"
password=XXXXXX  (Here is where I keep changing the password, no quotes, variations with or without caps)
; uncomment the next lines ipaddress and port with your local network info
; if you don't want the server to automatically find your 
; slingbox on the local network. Needed if server and slingbox are
; not on the same LAN segment. Also removes requirement for the netifaces module  
ipaddress=192.168.0.254
port=5001
; Repace with your finderid, If and only if you need remote access and don't  
;have a static ip and you plan on using my service to access you server remotely. 
; Please read the release notes. Most people don't need this.
;finderid=BAD95DAED609364C8204F2BC5FBDC363

;If you've got more than one slingbox set name. THe server will use this when generating 
; logs to make it easier to see what's going on. 
;name=something

; Following are the default values
; Default resolution when server starts. You can change this and reconnect without having to restart the server
; valid range = 1..15
;Resolution=12

; Valid Values 1, 6, 10, 15, 20, 30, 60   depending on resolution
;FrameRate=30

;Min 50 Max 8000
;VideoBandwidth=2000

; Min 0 (auto) max 63
;VideoSmoothness=63

; Send Iframe every n seconds Max 30
;IframeRate=5

; Audio Bit Rate. Valid Options 16, 20, 32, 40, 48, 64, 96 Default 32
;AudioBitRate=32

; Video Source  0, 1, 2, 3 depending on your hardware corrosponds to one of
;Composite, Component, HDMI or Tuner.
;ProHD 0=Tuner, 1=composite 2=svideo 3=component
;Solo, M1 0=composite 1=Svideo 2=component
;500  0=composite, 1=component, 2=HDMI
; If you don't set this the code will use the last configired input
; I recommend setting this value. If not set correctly you'll often only see a black screen. 
VideoSource = 3

[SERVER]
; local port number for the server to listen on for connections
port=8181
;maxstreams=10
;enableremote=yes

[REMOTE]
; see release notes for changing this if your remote doesn't work, But as a start make it 
;the same as your configured VideoSource 0-3. See above.
code=1
; path to an external remote control definitions file, if the defaults
; aren't working for you or you want to change the format of the web page.
; use the supplied remote.txt as a starting point.
;include=remote.txt

**Reply by JoeM2016** [reply]
Look for admin passwords in the Slingplayer Desk top log on pc's that were connected to the SB's. Mine are located at:
C:\Users\joe_m\AppData\Roaming\Sling Media\Slingplayer Desktop\Logs\console
open text file and search the file for
USER":"admin","PASSWORD
May also want to get:
FINDER_ID
The admin password is a 15-character alpha-numeric value

**Reply by PeteSki1618** [reply]
Thanks again for your reply.  I found the file and opened it but here's what I found:
"username":"admin","password":"E1:107802A0D90529889EC1FAXXXXXXXXXX"
I X'd out the last 10 characters but as you can see it is longer than 15 characters.  I tried it with or without the E1: with no luck.  Any other thoughts would be appreciated.

**Reply by JoeM2016** [reply]
Did you search for this exact text?  USER":"admin","PASSWORD  
My search finds {"USER":"admin","PASSWORD":"xxxxxxxxxxxxxxx"  x's are the 15-character alpha-numeric value.  This is a running log file, there should be multiple instances from each time the app was run.  Keep searching for additional instances that look similar to {"USER":"admin","PASSWORD":"xxxxxxxxxxxxxxx"

**Reply by waldorf-statler** [reply]
Apparently E1: gives the hashed password so for now no good. But I can see both that AND the 15-character lower case password, in different lines. Search for all instances of password...

**Reply by JoeM2016** [reply]
Example of a 15-character alpha-numeric value for the admin password needed for the config file.  yueSO3yF2Yd5In9.  Keep searching the file until you keep getting the same password.  The first instance may be a temporary one.

**Reply by PeteSki1618** [reply]
I appreciate all the time you have spent trying to help me.  I searched the entire file and cannot find anything that looks like what you describe.  Here is a sample of one of the findings:

setConnectionParamsToConfig...Box object : {"USER":"admin","PASSWORD":"admin","FINDER_ID":"030FEA9B34A8D74083A5FEXXXXXXXXXX","NAME":"My Slingbox","PRODUCT_ID":8,"IS_CONFIGURED":0,"IS_OWNER":true,"IS_SAVED":1}
23:49:27 352 : 5.0.0.83 : INFO :: DeLoreanLib :: If date field is not null then proceed to convertDateToUTC

I assume the password here is "admin" (no quotes in the config.exe file).  There are multiple instances with the same password and one other with one of the passwords I used to use upon signing in.  Tried them all but same results.  The Finder ID is correct.  Thanks again.

**Reply by JoeM2016** [reply]
Clip from my log;
  
{"USER":"admin","PASSWORD":"jAkxxxxxxxxxJMVc","FINDER_ID":"2C56xxxxxxxxxxxxxxxxxxxxxx71B1D70","NAME":"My Slingbox","PRODUCT_ID":18,"IS_CONFIGURED":1,"IS_OWNER":true,"IS_SAVED":1}
16:58:5 443 : 5.0.0.83 : INFO :: DeLoreanLib :: If date field is not null then proceed to convertDateToUTC

It is similar to yours except for the redacted 15-character password and finderid. Also, the product code is different. I am using a 500. I think I have seen a product code of 17 for 350's somewhere in this forum. Did you access your SB after N9 with the Desktop app? Were you able to connect to the SB before N9 with the app? In this forum I seen posts from followers that their admin password was admin after N9 when they were trying to reset their SB with the Desktop app. May want to look at the Decrypt Password #296 discussion, someone has a procedure for decrypting the 64-character password to the 15-character pw used by slinger.

**Reply by waldorf-statler** [reply]
I am also thinking that the fact that "IS_CONFIGURED" is set to 0 also implies it has been reset or bricked. Anyway as @JoeM2016 says, decoding the password from the E1 hash is covered elsewhere. The other approach is to check whether any of your other slingboxes are still operational / the password can be found as per discussion above.

**Reply by PeteSki1618** [reply]
Thank you guys so much for your replies.  Was out all day and haven't had a chance to look further into it.  I will check out the other threads about decrypting the passwords and see if that helps.  However, I have 3 boxes, all HDPros.  One box has only "admin" as the password listed, the other two have E1 numbers but one is 48 characters and the other 32.  NONE are 64???  I have not used the Desktop app in years so the information in it might be too old to consider.  I did not use the app after Nov9.  My friend has a laptop which had access to my box while it was up.  I will check his console log to see if he possibly has my information listed in it.  I will continue over to the #296 thread and see if anyone knows of a way to retrieve the password from the box itself.  It's a shame that I have 3 HDPros and even an old Slingbox Classic sitting here (I even tried that one with no luck) and none of them work.  Thank you guys SO much for all of your help and patience.  Will let you know if I have any success.

**Reply by JoeM2016** [reply]
Did your friend access the boxes after N9?

**Reply by PeteSki1618** [reply]
I'm not sure.  I know he tried just before N9 because I was having trouble and thought they shut down early but his was working.  Then mine was working.  He may have tried after N9 to see if it was down yet.  Why?  If he accessed after N9 does that automatically brick it?

**Reply by JoeM2016** [reply]
I have seen discussions that the boxes connected to the net after N9 are contacting sling servers, on their own, which downloads updates to the firmware which disables them.

**Comment by PeteSki1618** [comment]
I don't know if anyone is still monitoring this thread but if so, I would appreciate it if someone could offer me some help (again).  My M1 has been working great after all the help you guys gave me setting it up.  I am now trying to help a friend set up his ProHD and have everything working except I keep losing the Remote.  What worked for me was setting up a Static IP but for the Gateway use 127.0.0.1 so it doesn't go out onto the Internet and get bricked.  I did all of that on my friend's ProHD and all was working.  I then set up the batch file to automatically run the server upon boot up and as soon as I rebooted to test it out, I lost the remote.  The remote comes up in my browser and if I click on anything, the server does recognize the command so I see the Sending Key but nothing happens on the TV.  I haven't had time to try it but I am assuming that if I Reset the ProHD, go through the Unbricking process again, it will work but will probably be lost after a reboot.   I am a novice and don't understand a lot of things in the instructions but followed what all of you great guys suggested.  The only difference I can see between my system and my friend's is the router so one last piece of information:   My router is at 192.168.0.1 so when I set up the Static IP I used that for the IP, Subnet 255.255.255.0 and Gateway 127.0.0.1.  My friend's router is at 192.168.1.1 (notice 1.1 not 0.1).  Do I need to change the Gateway to 127.0.1.1 possibly?  His router also doesn't allow him to reserve an IP address so it may be a problem in the future if the IP of his server changes but for now I don't think it's the problem.  As always, thank you in advance for your help!

**Reply by GerryDazoo** [reply]
No, 127.0.0.1 is a local address it has nothing to do with the LAN subnet. Please post ALL the server output. I suspect you haven't configured the video source so the remote won't work. Do you see Keyscodes=[]? That the big clue to the issue.

**Comment by PeteSki1618** [comment]
Thanks for the prompt reply.  I don't have access to it now since it is at my friend's house but will post the results the next time I'm over there (probably next week).  However, I remember I have the Videosource=3 and under Remote I have code=3.   I tried all values 0-3 for both thinking that might be the problem.  And, in the Slingbox_server.exe box running on his server, when I click on a key in the remote I do see the keycodes appearing so I know the server is receiving the input from the remote, however, there is no response on the TV.  And, as I said, it was working fine both locally and remotely until I rebooted.  When I did the initial configuration in Slingplayer, it also worked.  I don't know what is changing.  Thanks again for your help.

**Reply by GerryDazoo** [reply]
I'm referring to the keycodes listed when you start up the video stream from your media player. Not when you click a button on the remote,

You don't need code=3 set if you have your videosource defined. It's only there if you want to use a remote config defined for a difference video source. You MUST have a active stream for the remote to work.

**Reply by PeteSki1618** [reply]
Ok thanks.  I'll delete that code and see what happens, however, it is working on MY system with the code=3 line and WAS working on his before the reboot but I will check it out the next time I go over there just to be sure.  Thanks again for all of your help.  It is VERY much appreciated!

**Comment by tensbum** [comment]
Newly retired federal government employee here with limited skillset in resolving home networking issues.  Based on information provided on this website, I believe I have configured my config.ini file correctly (see screenshot).  However, when I open VLC and attempt to run the stream, I receive a "ConnectionResetError: [Winerror 10054]" message.  

Can someone provide advice to resolve this issue?  Thanks a lot.

![Screenshot 2](https://user-images.githubusercontent.com/125129207/218256400-de6c0f1f-2a63-4734-9ad6-6db62ad98cd7.png)

**Reply by GerryDazoo** [reply]
You do not have the correct IP/Port for you slingbox. You've configures the [SERVER] port. 
Run the server without an IP address configured and it will probe your network and list all the slingboxes it finds. If you only have one box it will use that.

**Reply by tensbum** [reply]
Thanks for the prompt response.  Your suggestion resolved the WinError message. Now I have to resolve an admin PW issue.  Unfortunately, I do not know recall it.  If I am unable to locate it, am I SOL or is there a way to find it?  

![Screenshot 3](https://user-images.githubusercontent.com/125129207/218287745-9c52b29c-1c88-42c6-8911-656348346021.png)

**Reply by tensbum** [reply]
I was able to recover my SB M2 admin PW using this solution.  A big thanks to GerryDazoo and other community members for salvaging this useful network device.

https://github.com/GerryDazoo/Slinger/discussions/374

**Comment by PeteSki1618** [comment]
First I want to thank Gerry and all who have been helping getting my Slingboxes running again.  All have been running GREAT and use them EVERY day!  Two, hopefully quick, questions if you guys don't mind helping me with.  I have been using the HDS600rs.txt file for my remote and it has been working well except for the PgDn key.  Whenever I click on it I get an "Internal Server Error".  I don't know how to read the coding but noticed there was no key number after the 'PgDn' : command so I tried putting number 44 (since PgUp was 43) but got the same error.  My second question is I have to tune to channel 1001 almost every day.  It is tricky to get all 4 digits one at a time often either going to slowly or too fast.  Is there a way to add another button to tune to channel 1001?  As always, thank you in advance for your help!

**Reply by GerryDazoo** [reply]
Please post the server output when you get this error. 
Whoever made that remote left off the Channel field where you can just put in the channel number you want and press enter. 
 'Channel': Channel
The code will do the timing. Or you can add a Channel 1001 button like so
'Favch' : 9,18,18,9
Where the numbers are the keycodes for the digits and favch can be anything you want.

**Reply by PeteSki1618** [reply]
![Error](https://user-images.githubusercontent.com/117535328/221255103-9e1da1f1-1f1c-46da-b75d-6051405efca8.JPG)
Here is the error that I get when I press the PgDn key.  
Since I don't understand any of the programming, do I need to add any of the :&nbsp stuff or simply put in a line with 'Channel': Channel?  
Thanks again, as always!!!  You're the best!!!

**Reply by GerryDazoo** [reply]
Sorry, I meant the Slinger output. Add this to the end of the file
        'Fav' :9,18,18,9
        :&nbsp;&nbsp;
        'Channel': Channel

**Reply by PeteSki1618** [reply]
Super!  It all worked!!!  The PgDn key works by adding the key 44.  What I forgot is, because I'm away from home, I didn't shut down Slinger after editing the Remote Control file.  After shutting down Slinger, adding 44 after PgDn and adding your suggestion at the end of the file, I now have all of the functionality I need.  Thank you again so much for all of your help.  It is VERY much appreciated!!!

**Reply by PeteSki1618** [reply]
One quick question:  Is there any App like VLC that will run on Roku?  I have 3 TV's;  two have Firesticks but the third has Roku.  I could easily get another Firestick but would require getting an HDMI splitter since I don't have any more ports.  I tried the Roku Media Player and also Plex but can't find a way to enter the IP address of Slinger.  Thanks in advance as always.

**Reply by PeteSki1618** [reply]
Help again please!  I am away and my Slingbox was working fine.  All of a sudden I cannot connect and get an error message that says "8080 Stream Terminated for  66.231.199.254:50438 [WinError 10054] An existing connection was forcibly closed by the remote host".  Using Chrome Remote Desktop, I can see VLC connecting but then it gives me the above error.  I am posting the entire Slinger box if that helps.  Any help would be appreciated!!!  Thanks again in advance.

Version : 3.08d Running on Windows-10-10.0.19045-SP0 pid= 10352
Using config file config.ini
Connection Manager Running on port 8080 with 10 max streams using URL slingbox.
Streamer Running:  10 config.ini SLINGBOX 8080 8080 8388608
starting up on  port 8080 8388608
Checking for slingbox at ('192.168.1.3', 5001)
('192.168.1.3', 5001) OK
8080 Using slingbox at  ('192.168.1.3', 5001)
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
Reading Custom Remote definition from C:\Users\phils\Desktop\Slingbox\CustomRemotes\HDS600rsProHD.txt
 * Serving Flask app 'slingbox_server' (lazy loading)
 * Environment: production
   WARNING: This is a development server. Do not use it in a production deployment.
   Use a production WSGI server instead.
 * Debug mode: off
03/20/2023, 23:20:17.009   Streaming request from ('66.231.XXX.254', 50438)
Starting remote stream 1

Slinginfo  "Solo/Pro/ProHD" 12 30 192.168.1.3 5001 3000 10
8080 Starting Stream for  66.231.XXX.254:50438
Connecting... ('192.168.1.3', 5001) Control
8080 Selecting Video Source 3
Keycodes= [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 38, 39, 40, 41, 42, 43, 44, 45, 46, 47, 48, 49, 53, 56, 57, 59, 60, 62, 63, 64, 69]
VideoParameters: Resolution= 12 FrameRate= 30 VideoBandwidth= 2000 VideoSmoothness= 63 IframeRate= 5 AudioBitRate= 64
Connecting... ('192.168.1.3', 5001) Stream
8080 Stream started at 03/20/2023, 23:20:23.819  733 0
.03/20/2023, 23:20:42.990  8080 Stream Terminated for  66.231.XXX.254:50438 [WinError 10054] An existing connection was forcibly closed by the remote host
0 active remote connections
8080 Shutting down connections
8080 Logging Out
Streamer:  8080 Waiting for first stream, flushing any IR requests that arrive while not connected to slingbox
